feat: secondary activation flow

kappa90 opened the pull request:
## Problem
See https://github.com/PostHog/meta/pull/65

We want to push users to complete secondary actions after the main ingestion flow

## Changes
- Add a secondary sidebar that is activated after the main ingestion flow
- Users can complete secondary tasks, for now:
    - Ingest an event: redirects to the main ingestion flow
    - Invite a team member: opens the invitation modal
    - Setup session recordings: shows the session recordings settings in the sidebar
    - Track custom actions: redirects them to the right doc (will improve this in a second iteration so they don't have to leave the app), and tracks if there's any custom event definition
    - Install an app: sends them to the apps marketplace
- Tasks can be skipped (list of skipped tasks is saved locally in local storage, which means logging in from a different device will show again the skipped tasks as uncompleted, good enough for v1)
- The sidebar reminder on the top bar disappears once all tasks are completed or skipped
- This sidebar can always be found again from the Help Button
- Substitutes the prompts based activation flow, which had a CTA to track marketing websites, in a second iteration we'll include a way to open the docs in the sidebar, and show additional non-blocking tasks such as this one
- Bonus: there's a cute professor hedgehog

## Design
I put the sidebar on the right as the small button in the topbar couldn't be fit anywhere else.
 
<img width="526" alt="image" src="https://user-images.githubusercontent.com/7335343/203850658-3c8fce50-f712-4a50-8033-35e9ea0e90f9.png">

<img width="526" alt="image" src="https://user-images.githubusercontent.com/7335343/203850006-34287c26-5fb9-4284-8123-4fe035612a33.png">

## Copy
This is the buttons copy (tbd):
```
- Ingest your first event
     - Ingest your first event to get started with PostHog
- Invite a team member
     - Every person in your company can benefit from PostHog
- Setup session recordings
     - See how your users are using your product
- Track custom events
     - Track custom events to get more insights into your product
- Install your first app
     - Extend PostHog's core functionality with apps
```


👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?
- Added a small logic test

kappa90 committed:
feat: secondary activation flow

kappa90 committed:
fix position

kappa90 committed:
add close button

kappa90 committed:
add logic test

kappa90 committed:
remove unused test

kappa90 commented:
@mariusandra I renamed a logic from `ActivationLogic` to `activationLogic` but the old logic name is still picked up and the tests fail, do you know what's going on?

alexkim205 commented:
> @mariusandra I renamed a logic from `ActivationLogic` to `activationLogic` but the old logic name is still picked up and the tests fail, do you know what's going on?

Have you tried to regenerate the type files by running `yarn typegen:clean`?

alexkim205 commented on frontend/src/lib/components/ActivationSidebar/ActivationSidebar.tsx:
```diff
@@ -0,0 +1,126 @@
+import { SessionRecording as SessionRecordingConfig } from 'scenes/project/Settings/SessionRecording'
+import { ProfessorHog } from '../hedgehogs'
+
+const Task = ({ id, name, description, completed, canSkip, skipped, url }: Task): JSX.Element => {
```
Rename `Task` to `TaskProps` or something different so that we don't redefine Task again

alexkim205 commented on frontend/src/lib/components/ActivationSidebar/ActivationLogic.ts:
```diff
@@ -0,0 +1,238 @@
+
+import type { activationLogicType } from './activationLogicType'
+
+export enum ACTIVATION_TASKS {
```
Nit: we usually don't capitalize enums, and should probably follow the standard here as well

joethreepwood commented on frontend/src/lib/components/ActivationSidebar/ActivationLogic.ts:
```diff
@@ -0,0 +1,238 @@
+                            tasks.push({
+                                id: ACTIVATION_TASKS.INVITE_TEAM_MEMBER,
+                                name: 'Invite a team member',
+                                description: 'Every person in your company can benefit from PostHog',
```
Users may not be in a company. They could be in some other sort of organization. We could say team? But then that may be too limiting.

```suggestion
                                description: 'Everyone in your organization can benefit from PostHog',
```

joethreepwood commented on frontend/src/lib/components/ActivationSidebar/ActivationSidebar.tsx:
```diff
@@ -0,0 +1,126 @@
+                ) : (
+                    <>
+                        <h2 className="subtitle">Quick Start</h2>
+                        <p>Learn how to get the most out of PostHog, straigth away.</p>
```
```suggestion
                        <p>Learn how to get the most out of PostHog, straight away.</p>
```

joethreepwood commented on frontend/src/lib/components/ActivationSidebar/ActivationSidebar.tsx:
```diff
@@ -0,0 +1,126 @@
+                                    percent={completionPercent}
+                                    format={() => activeTasks.length}
+                                />
+                                <p className="text-muted mt-2">Remaining tasks</p>
```
It may be worth taking the word 'task' out, just so it doesn't feel so choreish?

"Still to go" maybe?

joethreepwood commented on frontend/src/lib/components/ActivationSidebar/ActivationSidebar.tsx:
```diff
@@ -0,0 +1,126 @@
+                        </div>
+                        {activeTasks.length > 0 && (
+                            <div className="mt-4">
+                                <div className="text-muted uppercase text-xs">Next tasks</div>
```
In keeping with tasks being choreish, maybe: "What's next?"

joethreepwood commented on frontend/src/lib/components/HelpButton/HelpButton.tsx:
```diff
@@ -145,19 +147,17 @@ export function HelpButton({
+                                hideHelp()
+                            }}
+                        >
+                            Quick Start using PostHog
```
```suggestion
                            Quick Start
```
'Using' just sounds weird here to me. And it's a little redundant.

joethreepwood reviewed (approved):
Some language tweaks, but lgtm.

kappa90 committed:
review changes

kappa90 commented:
- Fixed the copy and code based on @alexkim205 and @joethreepwood's reviews (thank you!)
- Added a tracking event when this sidebar is opened showing how many tasks have been completed
- Fixed the reducer for skipped tasks to be per-project, rather than per-user

kappa90 commented:
> > @mariusandra I renamed a logic from `ActivationLogic` to `activationLogic` but the old logic name is still picked up and the tests fail, do you know what's going on?
> 
> Have you tried to regenerate the type files by running `yarn typegen:clean`?

That didn't seem to work 😢

alexkim205 committed:
Merge remote-tracking branch 'origin' into feat/secondary-onboarding

alexkim205 committed:
refactor logics and fix

alexkim205 commented:
> > > @mariusandra I renamed a logic from `ActivationLogic` to `activationLogic` but the old logic name is still picked up and the tests fail, do you know what's going on?
> > 
> > 
> > Have you tried to regenerate the type files by running `yarn typegen:clean`?
> 
> That didn't seem to work 😢

Pushed some changes that should fix this (I hope)!

alexkim205 committed:
use urls don't construct url manually

kappa90 commented on frontend/src/lib/components/ActivationSidebar/activationLogic.ts:
```diff
@@ -245,20 +248,10 @@ export const activationLogic = kea<activationLogicType>([
             actions.loadCustomEvents()
         },
     })),
-    urlToAction(({ actions, values }) => ({
```
Why did you remove this?

kappa90 committed:
restore urlToAction behavior

raquelmsmith commented on frontend/src/lib/components/ActivationSidebar/ActivationSidebar.tsx:
```diff
@@ -0,0 +1,126 @@
+    const content = (
+        <div className="my-4 mx-2">
+            <p className="m-0">{name}</p>
+            {!completed && !skipped && <p className="font-normal text-xs mt-2 mb-0">{description}</p>}
```
For whatever reason all `<p>` tags have a 5px left margin. We should remove it here since it looks oddly indented.

<img width="389" alt="image" src="https://user-images.githubusercontent.com/18598166/204102715-7e406bdb-5423-4971-b492-a863eb7bd0f5.png">

raquelmsmith commented on frontend/src/lib/components/ActivationSidebar/ActivationSidebar.tsx:
```diff
@@ -0,0 +1,126 @@
+                ) : (
+                    <>
+                        <h2 className="subtitle">Quick Start</h2>
+                        <p>Learn how to get the most out of PostHog, straight away.</p>
```
The `, straight away.` reads weird to me, and I (as a user) am not sure what it means to "get the most out of PostHog". I'd reword to something like...

```suggestion
                        <p>Use our Quick Start guide to learn about everything PostHog can do for you and your product.</p>
```

raquelmsmith commented on frontend/src/layout/navigation/navigationLogic.ts:
```diff
@@ -23,6 +23,9 @@ export const navigationLogic = kea<navigationLogicType>({
     actions: {
         toggleSideBarBase: true,
         toggleSideBarMobile: true,
+        toggleActivationSideBar: true,
+        showActivationSideBar: true,
+        hideActivationSideBar: true,
```
It seems a little unnecessary to have a `toggle` plus a `hide` and a `show`. Theoretically we can just do with a `toggle`, yes?

kappa90 replied:
There's certain places where I always want to either show or hide, so it's the toggle that's just an extra, I'd just keep them, since it's just kea actions

raquelmsmith commented on frontend/src/lib/utils/eventUsageLogic.ts:
```diff
@@ -480,6 +480,11 @@ export const eventUsageLogic = kea<eventUsageLogicType>({
         reportInstanceSettingChange: (name: string, value: string | boolean | number) => ({ name, value }),
         reportAxisUnitsChanged: (properties: Record<string, any>) => ({ ...properties }),
         reportTeamSettingChange: (name: string, value: any) => ({ name, value }),
+        reportActivationSideBarShown: (
```
I think it would also be good to record events for when people click the different actions within the sidebar so we can track how well the sidebar works for getting people to do the different things.

kappa90 replied:
Makes sense, done!

raquelmsmith commented on frontend/src/lib/components/ActivationSidebar/activationLogic.ts:
```diff
@@ -0,0 +1,266 @@
+import type { activationLogicType } from './activationLogicType'
+import { urls } from 'scenes/urls'
+
+export enum ActivationTasks {
```
We should def add tasks for creating a dashboard, viewing an insight, etc. Most people are signing up for product analytics and we want to make sure they are actually getting a good tour of that product before we push them onto something else like recordings, esp since our activation metric depends on that.

kappa90 replied:
Done done done

raquelmsmith reviewed (changes requested):
Love it! Few comments, but overall excited to see how this works. 

Have we decided how we are planning to define success/failure of this?

kappa90 committed:
review changes

kappa90 committed:
forgot to commit code doh

kappa90 committed:
fix

kappa90 committed:
fix

kappa90 committed:
Merge branch 'master' into feat/secondary-onboarding

kappa90 committed:
fix

raquelmsmith reviewed (approved):
:shipit:

kappa90 commented:
Regarding considering this a success or not, we should check if individual CTA push to an increase in the related metric (discoveries, plugins installed, session recording on, etc.)

kappa90 committed:
add ff for experiment

kappa90 committed:
fix cypress test

kappa90 committed:
Merge branch 'master' into feat/secondary-onboarding

kappa90 committed:
fix cypress

kappa90 committed:
change color
